### PR TITLE
Simplify demo template

### LINKED ISSUE #2225: Flutter default sample defines key values that are not used.
In the Flutter default sample (sourced here: `flutter/packages/flutter_tools/templates/create/lib/main.dart.tmpl`), the sample code includes the following statements:

key: const ValueKey('counter')
key: const ValueKey('fab'),

Neither of these keys are used, and these statements should probably be removed from the sample.

### Steps to Reproduce

1. Run 'flutter create myapp'.

### Flutter Version
Flutter from https://github.com/flutter/flutter.git (on hackathon)
flutter revision: 4860e63f444cf3786a6b136430f3058a7b851041 (2 days ago)
engine revision : a53faaf7983025f83a2059021a409c6e98beb58d

[✓] The Android toolchain is fully installed.
[✓] The Atom development environment is fully installed.

### PR DESCRIPTION
**Pull request by abarth**
We don't need these keys. Also, improve style in several places.

Fixes #2225

### CODE REVIEW

**Comment by abarth** [comment]
@qchong @collinjackson

File: packages/flutter_tools/templates/create/lib/main.dart.tmpl [hunk 1]
```diff
@@ -30,17 +30,13 @@ class _FlutterDemoState extends State<FlutterDemo> {
         center: new Text('Flutter Demo')
       ),
       body: new Center(
-        child: new Text(
-          'Button tapped $_counter times.',
-          key: const ValueKey('counter')
-        )
+        child: new Text('Button tapped $_counter times.')
```
Comment (Hixie) [inline packages/flutter_tools/templates/create/lib/main.dart.tmpl]: `time${ _counter == 1 ? '' : 's' }`

File: packages/flutter_tools/templates/create/lib/main.dart.tmpl [hunk 2]
```diff
@@ -30,17 +30,13 @@ class _FlutterDemoState extends State<FlutterDemo> {
...
       ),
       floatingActionButton: new FloatingActionButton(
-        key: const ValueKey('fab'),
+        onPressed: _incrementCounter,
         child: new Icon(
           icon: 'content/add'
```
Comment (Hixie) [inline packages/flutter_tools/templates/create/lib/main.dart.tmpl]: tooltip?

**Comment by Hixie** [comment]
lgtm

**Comment by abarth** [comment]
PTAL

File: packages/flutter_tools/templates/create/lib/main.dart.tmpl [hunk 3]
```diff
@@ -30,17 +30,14 @@ class _FlutterDemoState extends State<FlutterDemo> {
         center: new Text('Flutter Demo')
       ),
       body: new Center(
-        child: new Text(
-          'Button tapped $_counter times.',
-          key: const ValueKey('counter')
```
Comment (yjbanov) [inline packages/flutter_tools/templates/create/lib/main.dart.tmpl]: This key is used by the driver test to locate the text

File: packages/flutter_tools/templates/create/lib/main.dart.tmpl [hunk 4]
```diff
@@ -30,17 +30,14 @@ class _FlutterDemoState extends State<FlutterDemo> {
...
-        )
+        child: new Text('Button tapped $_counter time${ _counter == 1 ? '' : 's' }.')
       ),
       floatingActionButton: new FloatingActionButton(
-        key: const ValueKey('fab'),
```
Comment (yjbanov) [inline packages/flutter_tools/templates/create/lib/main.dart.tmpl]: this key served the same purpose

File: packages/flutter_tools/templates/create/lib/main.dart.tmpl (hunk 3 above)
Comment (abarth) [inline packages/flutter_tools/templates/create/lib/main.dart.tmpl]: I don't think we should junk up the main example code with extra stuff for the driver test.

**Comment by abarth** [comment]
Spoke with @yjbanov .  The plan is to use the tooltip for the driver test, which promotes its use for accessibility as well.

**Comment by qchong** [comment]
Thanks for the updates, @abarth. Can you please merge these changes into the 'hackathon' branch also?

**Comment by abarth** [comment]
Done.

**Comment by Hixie** [comment]
still lgtm

**Pull request merged**